File: src/backend/search-server/backend/src/models/music.py
from pydantic import BaseModel

from .album import AlbumNested, Album
from .artist import Artist

class Music(BaseModel):
    id: str
    title: str
    playTime: str
    album: Album
    artist: Artist

class MusicIndexRequest(BaseModel):
    id: str
    title: str
    playTime: str
    album: AlbumNested
    artist: Artist

class MusicNested(BaseModel):
    id: str
    title: str
    playTime: str
    album: AlbumNested
    artist: Artist

File: src/backend/domain-module/music/src/main/java/com/lalala/music/domain/Music.java
package com.lalala.music.domain;

import java.time.LocalDateTime;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class Music {
    private final Long id;
    private final String title;
    private final short playTime;
    private final Album album;
    private final File file;
    private final Artist artist;
    private final LocalDateTime createdAt;
    private final LocalDateTime updatedAt;
}
